kube/src/api/gvk.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{Error, Result};
+use serde::{Deserialize, Serialize};
+
+
+/// Contains enough information to identify API Resource.
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct GroupVersionKind {
+    /// API group
+    pub group: String,
+    /// Version
+    pub version: String,
+    /// Kind
+    pub kind: String,
+}
+
+impl GroupVersionKind {
+    /// Set the api group, version, and kind for a resource
+    pub fn gvk(group_: &str, version_: &str, kind_: &str) -> Result<Self> {
+        let version = version_.to_string();
+        let group = group_.to_string();
+        let kind = kind_.to_string();
+        if version.is_empty() {
```

Comment:
Aside: this validation is incomplete anyway (it's impossible to validate input without asking apiserver). Also, it **seems** to me that typo in the resource name is more likely than providing an empty string, and Rust in general discourages using empty strings as default values.

That's why I'd consider removing these checks and making this constructor infallible.

Reply:
Ok, if we're not doing `TryFrom`, then it's probably better to drop the checks as you suggested. All the values are either generated via `kube-derive` or returned from the apiserver anyway.